datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -3870,6 +3873,39 @@ digraph {
         );
     }
 
+    #[test]
```

Comment:
Could you also please add a sqllogictest? Perhaps by extending https://github.com/apache/datafusion/blob/main/datafusion/sqllogictest/test_files/distinct_on.slt ?